### ISSUE TITLE
Internal Error: Invalid fork construction

### ISSUE THREAD

**Issue by kunicmarko20** [maciejhirsz/logos#74]
I am getting this error and I have no idea what am I doing wrong:

```
use logos::Logos;

#[derive(Logos, Debug, PartialEq)]
enum Token {
    #[end]
    End,
    #[error]
    Error,
    #[regex = r"(class|interface|trait)\s++(?P<name>[a-zA-Z_\x7f-\xff:][a-zA-Z0-9_\x7f-\xff:\-]*)"]
    Element,
    #[regex = r"namespace\s++(?P<namespace>[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+(?:\s*+\\\s*+[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*+)*+)?\s*+[\{;]"]
    Namespace,

}
```

I guess my regex isn't appropriate?

**Comment by maciejhirsz** [comment]
I'm not sure what you are trying to achieve with `\s++` or `\s*+`, you should be using either `+` or `*`. That isn't really the problem here.

It also seems to me that you are trying to pack a full parser into the lexer, instead of building the parser on top of it. `[a-zA-Z_\x7f-\xff:][a-zA-Z0-9_\x7f-\xff:\-]*` should be it's own token, the lexer by default skips whitespace between tokens so you don't have to write `\s+` everywhere. You won't be able to retrieve the named subsections from the output (`?P<namespace>` will do nothing).

All that said, the error you are getting is an internal error and it should be, at very least, more informative.

**Comment by kunicmarko20** [comment]
Yep, I understood later that I want a parser and a lexer, I am lacking knowledge. Would be nice if you could make errors more descriptive.

Thank you for your reply.

**Comment by maciejhirsz** [comment]
I'd keep this open if you don't mind. This definitely needs a book explaining what this whole crate is for :).

I'm able to reproduce the invalid fork with `#[regex = r"aa*+"]`, so that should at least be easy to track and add the error.

**Comment by maciejhirsz** [comment]
This should be now fixed in 0.10.0-rc3, please re-open if it's not the case.
